Reference/google/protobuf/compiler/plugin.pb.swift

```diff
@@ -79,7 +79,17 @@ public struct Google_Protobuf_Compiler_CodeGeneratorRequest: ProtobufGeneratedMe
   public var fileToGenerate: [String] = []
 
   ///   The generator parameter passed on the command-line.
-  public var parameter: String? = nil
+  private var _parameter: String? = nil
+  public var parameter: String {
+    get {return _parameter ?? ""}
+    set {_parameter = newValue}
+  }
+  public var hasParameter: Bool {
+    return _parameter != nil
+  }
+  public mutating func clearParameter() {
+    return _parameter = nil
+  }
```

Comment:
Thanks for your careful response.  I'll continue to think about it.

To be clear, my only question at this point is with proto2 optional fields that do not have an explicit default.  I agree with you about the other cases.  In particular, I agree that your generation of proto2 optional fields with explicit defaults is the only appropriate way to go.  I agree that we need `has` methods for message-typed fields.

A couple of your comments above intrigue me:

> ... even if we have to wait for a future version of Swift to improve on this ...

I haven't given much thought to this, but do you think we have enough versioning capabilities to make this kind of breaking change to our API in the future?

> It's reasonable to omit has/clear for proto3 because the defaults are well known...

Actually, it's reasonable to omit them because proto3 does not guarantee access to field presence information.  Proto2 does guarantee this.

(As you quoted just above, the defaults for proto2 are also well-known.)

> The large number of force-unwraps that I removed in the generator code after I updated descriptor.pb.swift to use non-optional properties is a sign to me that the user-facing API was not ideal to work with.

In my opinion, this is your strongest argument.  My primary concern is with usability in real code.  I'll have to take a careful look at those changes.